### ISSUE TITLE
Apigeecli doesn't conform to GCP Apigee naming conventions with OAS

### ISSUE THREAD

**Issue by buildwithjz** [apigee/apigeecli#145]
When using an OpenAPI spec to create an API proxy revision in the GCP Apigee UI (as per the documentation [here](https://cloud.google.com/apigee/docs/api-platform/tutorials/create-api-proxy-openapi-spec#createtheapiproxy)), Apigee populates fields in the following way:
```
Name: title property from the OpenAPI Specification with spaces replaced by dashes
Base path: Name field content converted to all lower case
Target (Existing API): servers property from the OpenAPI Specification
```
This is proven to be true when creating a proxy from the [petstore apispec](https://github.com/apigee/apigeecli/blob/main/test/petstore.yaml)

<img width="1352" alt="Screen Shot 2023-01-23 at 2 15 47 PM" src="https://user-images.githubusercontent.com/82625873/213953913-ddc6024a-90b9-4b46-a235-97b029efd3a7.png">

However when using the same OpenAPI spec using `apigeecli` in the following way:

```
$ apigeecli apis create openapi -f petstore.yaml -o $org -t $token -n OpenAPI-Petstore
```
The following settings are populated into apigee

<img width="1625" alt="Screen Shot 2023-01-23 at 2 38 39 PM" src="https://user-images.githubusercontent.com/82625873/213954133-36914235-9e37-4b98-9ef9-97f44372149d.png">

It seems that `apigeecli` instead strips the path in the URL provided in `servers` and uses it as the base path of the API proxy, while using the hostname (with no path) of that same URL and uses that as the target endpoint for that proxy (unless overridden by `--target-url`. 

I want to know whether this departure from the GCP Apigee X UI is a bug, or whether this is expected behaviour. 

My interpretation is that `apigeecli` treats the `servers` field in OAS more as a definition of its basepath, rather than a definition of its target endpoint. Target endpoints/URLs are instead designed to be specified in `--target-url`. If this is the case, I would like to see documentation clarifying on this behaviour.

Thanks!

**Comment by srinandan** [comment]
Thank you for opening the issue. There are two parts to the question
1. What should the basePath be?
2. What should the target url be?

For 1, I think apigeecli's implementation is correct. The basePath is provided by the user in accordance with the [spec](https://swagger.io/docs/specification/api-host-and-base-path/). It is also possible that only the basePath is added to servers, like this:
```
servers:
  - url: /
```

Therefore, in your example, the basePath is correctly interpreted as `/v2`. 

For 2, I think Apigee UI's implementation is correct. It does make sense to have the target endpoint as `https://petstore.swagger.io/v2`.  I will make the change.

**Comment by srinandan** [comment]
The fix is available in v1.117, closing the issue.
